### PR TITLE
Proposal for a release workflow

### PR DESCRIPTION
**Pull request by cdr-chakotay**
Proposal for a release workflow using Gradle and GH Actions. Similar to Transloadit Java SDK.

Uses a version.txt to obtain the current project version. 

Let me know, what you think :)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -22,49 +28,79 @@ dependencies {
...
-                    license {
-                        name 'The MIT License (MIT)'
-                        url 'http://opensource.org/licenses/MIT'
-                    }
-                }
+artifacts {
+    archives sourcesJar, javadocJar
+}
+
+def versionNo = rootProject.file('version.txt').text.trim()
```
Comment (Acconut) [inline build.gradle:44]: Can we use the same version.properties file as in the Java SDK? i.e. https://github.com/transloadit/java-sdk/blob/807e997e48c1697cf557bfb1d97a612ac62e7866/build.gradle#L14, to bring the two scripts closer together

File: build.gradle [hunk 2]
```diff
@@ -1,3 +1,9 @@
+plugins {
+    id 'signing'
+    id("io.github.gradle-nexus.publish-plugin") version "1.1.0"
```
Comment (Acconut) [inline build.gradle:3]: Please use a consistent quotation style in the files. It seems we have more single quotes than double quotes, so let's go with single quotes.

File: build.gradle [hunk 3]
```diff
@@ -22,49 +28,79 @@ dependencies {
...
+    inceptionYear '2015'
+    licenses {
+        license {
+            name 'The MIT License (MIT)'
+            url 'http://opensource.org/licenses/MIT'
+        }
+    }
+}
+
+publishing{
```
Comment (Acconut) [inline build.gradle:73]: One space before the bracket, please :)

File: build.gradle [hunk 4]
```diff
@@ -22,49 +28,79 @@ dependencies {
...
+                def root = asNode()
+                root.appendNode('description', 'Java client for tus, the resumable file uploading protocol.')
+                root.children().last() + pomConfig
             }
-        }.writeTo("build/libs/pom.xml")
+        }
+    }
+}
+
+signing{
```
Comment (Acconut) [inline build.gradle:91]: One space before the bracket, please :)

**Review by Acconut — CHANGES_REQUESTED** [review]
Thank you! This looks very promising and I only got a few comments regarding code format and the version file. If I understood correctly, you already set up the secrets in the repository, didn't you?

**Comment by cdr-chakotay** [comment]
Yes I've already added the secrets.

We could use the same version file, I just wanted to avoid the config slurper. But I can add this

File: build.gradle (hunk 1 above)
Comment (cdr-chakotay) [reply build.gradle:44]: Yes

**Review by Acconut — APPROVED** [review]
LGTM! Let get this rolling! Thank you!

**Comment by Acconut** [comment]
Also, would you be able to see if there any old releases of tus-java-client missing on Maven Central (as is the case for tus-android-client: https://github.com/tus/tus-android-client/issues/45)?

**Comment by cdr-chakotay** [comment]
> Also, would you be able to see if there any old releases of tus-java-client missing on Maven Central (as is the case for tus-android-client: [tus/tus-android-client#45](https://github.com/tus/tus-android-client/issues/45))?

We are starting with 0.4.0 on Maven. Older releases are not available there. Honestly searching through Bintray is no fun. They even shut down the website and search already.

**Comment by Acconut** [comment]
> Older releases are not available there. Honestly searching through Bintray is no fun. They even shut down the website and search already.

Ok, if it is too much hassle, we can ignore about the old releases for now

**Pull request merged**